src/lib/journey/journey.store.ts

```diff
@@ -139,32 +144,113 @@ export async function initialize(journey?: string): Promise<InitObject> {
        * --------------------------------------------------------------------
        * Details: Now that we have a new authId (the identification of the
        * fresh step) let's populate this new step with old callback data if
-       * the stage is the same. If not, the user will have to refill form. We
+       * allowed with this stage. If not, the user will have to refill form. We
        * will display the error we collected from the previous submission,
        * restart the flow, and provide better UX with the previous form data,
        * so the user doesn't have to refill the form.
        ******************************************************************* */
-      if (restartStep.type === StepType.Step && restartStep.getStage() === previousStage) {
-        if (previousCallbacks) restartStep.callbacks = previousCallbacks;
-        restartStep.payload = {
-          ...previousPayload,
-          authId: restartStep.payload.authId
-        };
+      if (restartedStep.type === StepType.Step) {
+        const currentStage = restartedStep.getStage() || '';
+        // TODO: Convert hardcoded stage of 'registration' to a configurable allowlist
+        if (currentStage === previousStage && currentStage.toLowerCase().includes('registration')) {
+          if (previousCallbacks) {
+            // Iterate over the callbacks to find any password and reset to empty value
+            restartedStep.callbacks = previousCallbacks.map(
+              (callback) => {
+                callback.setInputValue('');
+                return callback;
+              },
+            );
+            restartedStep.payload = {
+              ...previousPayload,
+              authId: restartedStep.payload.authId,
+            };
+          }
+        }
```

Comment:
All of this is going to be removed. I think this is legacy code that I added a long time ago due to a bad ID Cloud tenant.

Reply:
This block of code has been removed and will be included in the update.